Improve validations wording in Active Record Basics guide [ci-skip] (#46925)
rails/rails: guides/source/active_record_basics.md

models and validate that an attribute value is not empty, is unique and not
already in the database, follows a specific format, and many more.

Methods like `save`, `create` and `update` validate a model before persisting
it to the database. When a model is invalid these methods return `false` and no

> database operation gets performed. All of these methods have a bang counterpart

Reviewer: ```suggestion
database operation will be performed. All of these methods have a bang counterpart
```

Reviewer: Or maybe "database operations are performed" in the plural because there may be more than one operation.

Author: Thanks. Went with "database operations are performed".

(that is, `save!`, `create!` and `update!`), which are stricter in that they

> raise the exception `ActiveRecord::RecordInvalid` if validation fails.

Reviewer: ```suggestion
raise the exception `ActiveRecord::RecordInvalid` when validation fails.
```

Author: I've also changed the wording of exception a bit.
```suggestion
raise an `ActiveRecord::RecordInvalid` exception when validation fails.
```
